cpp/daal/src/algorithms/dtrees/dtrees_model.cpp

```diff
@@ -100,7 +108,7 @@ bool ModelImpl::reserve(const size_t nTrees)
     _probTbl.reset(new DataCollection());
     _probTbl->resize(nTrees);
 
-    return _serializationData.get();
+    return isValid();
```

Comment:
It would be also good to do the following modifications:
`if (!_probTbl->resize(nTrees)) return false;`